Analysis/Tasks/PWGHF/D0D0barCorrelator.cxx

```diff
@@ -0,0 +1,850 @@
+        }
+        if (cutYCandMax >= 0. && std::abs(RecoDecay::Y(array{particle2.px(), particle2.py(), particle2.pz()}, RecoDecay::getMassPDG(particle2.pdgCode()))) > cutYCandMax) {
+          continue;
+        }
+        if (cutPtCandMin >= 0. && std::abs(particle2.pt()) < cutPtCandMin) {
+          continue;
+        }
+        if (particle2.pdgCode() == particle1.pdgCode()) {
+          //check whether mothers of quark cbar (from associated loop) are still '-4' particles - in that case the cbar quark comes from its own fragmentation, skip it
+          if (particlesMC.iteratorAt(particle2.mother0()).pdgCode() == particle2.pdgCode()) {
+            continue;
+          }
+          //Excluding self-correlations
+          if (particle1.mRowIndex == particle2.mRowIndex) {
+            continue;
+          }
+          entryD0D0barPair(getDeltaPhi(particle2.phi(), particle1.phi()),
+                           particle2.eta() - particle1.eta(),
+                           particle1.pt(),
+                           particle2.pt());
+        } // end outer if (check PDG associate)
+      }   // end inner loop
+    }     //end outer loop
+    registry.fill(HIST("hcountCCbarPerEvent"), counterccbar);
+    registry.fill(HIST("hcountCCbarPerEventPreEtaCut"), counterccbarPreEtasel);
+  }
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  WorkflowSpec workflow{};
+  const bool doMCccbar = cfgc.options().get<bool>("doMCccbar");
+  const bool doMCGen = cfgc.options().get<bool>("doMCGen");
+  const bool doMCRec = cfgc.options().get<bool>("doMCRec");
+  const bool doLikeSign = cfgc.options().get<bool>("doLikeSign");
+  if (!doLikeSign) { //unlike-sign analyses
+    if (doMCGen) {   //MC-Gen analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelatorMCGen>(cfgc, TaskName{"d0d0bar-correlator-mc-gen"}));
+    } else if (doMCRec) { //MC-Reco analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelatorMCRec>(cfgc, TaskName{"d0d0bar-correlator-mc-rec"}));
+    } else if (doMCccbar) { //MC-Reco analysis
+      workflow.push_back(adaptAnalysisTask<CCbarCorrelatorMCGen>(cfgc, TaskName{"ccbar-correlator-mc-gen"}));
+    } else { //data analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelator>(cfgc, TaskName{"d0d0bar-correlator"}));
+    }
+  } else {         //ike-sign analyses
+    if (doMCGen) { //MC-Gen analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelatorMCGenLS>(cfgc, TaskName{"d0d0bar-correlator-mc-gen-ls"}));
+    } else if (doMCRec) { //MC-Reco analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelatorMCRecLS>(cfgc, TaskName{"d0d0bar-correlator-mc-rec-ls"}));
+    } else if (doMCccbar) { //MC-Reco analysis
+      workflow.push_back(adaptAnalysisTask<CCbarCorrelatorMCGenLS>(cfgc, TaskName{"ccbar-correlator-mc-gen-ls"}));
+    } else { //data analysis
+      workflow.push_back(adaptAnalysisTask<D0D0barCorrelatorLS>(cfgc, TaskName{"d0d0bar-correlator-ls"}));
+    }
```

Comment:
Name the tasks such that `TaskName` is not necessary. Include the prefix `Hf`.

Reply:
Renamed the tasks, anyway I left the TaskName, as I also find in theother HF tasks, because otherwise the automatic dashing of the names creates bad names as hf-d-0-d-0-bar-correlations etc.

Reply:
It puts hyphens around numbers? That's bad.

Reply:
Yes, from what I remember, I had already this when Jan Fiete asked for this on mattermost in March : I had already modified the code in that sense, but then I preferred to revert back to TaskName seeing unexpected names - don't know if things have changed though